gnomad_qc/v5/data_ingestion/federated_validity_checks.py

```diff
@@ -14,6 +20,97 @@
 logger.setLevel(logging.INFO)
 
 
+def check_missingness(
+    ht,
+    missingness_threshold: float = 0.5,
+    struct_annotations: list = ["grpmax", "fafmax", "histograms"],
+    indexed_array_annotations: dict = {
+        "faf": "faf_index_dict",
+        "freq": "freq_index_dict",
+    },
+) -> None:
+    """
+    Check for fraction of missing data in the Table
+
+    :param ht: Input Table.
+    :param missingness_threshold: Upper cutoff for allowed amount of missingness. Default is 0.50.
+    :param struct_annotations: List of struct annotations to check for missingness. Default is  ['grpmax', 'fafmax', 'histograms'].
+    :param indexed_array_annotations: Dictionary of indexed arrays annotations to check for missingnes, where keys are the names of the annotation, and values are the names of the globals containing the mapping of group name to index for that key. Default is {'faf':'faf_index_dict', 'freq':'freq_index_dict'}.
+    :return: None
+    """
+
+    logger.info("Checking for missingness within struct annotations...")
+    # Determine missingness of each struct annotation.
+    metric_missingness = {}
+    for metric in struct_annotations:
+        metric_missingness.update(check_missingness_of_struct(ht[metric], metric))
+
+    missingness_struct = ht.aggregate(hl.struct(**metric_missingness))
+    missingness_dict = flatten_missingness_struct(missingness_struct)
+
+    logger.info("Checking for missingness within indexed array annotations...")
+    # Determine missingness of each indexed array annotation.
+    # missingness_dict.update(check_array_struct_missingness(ht))
+    missingness_dict.update(check_array_struct_missingness(ht))
+
+    # Report whether or not each metric pass or fails the missingness check
+    # based on the missingness_threshold.
+    for field, missingness in missingness_dict.items():
+        if missingness > missingness_threshold:
+            logger.info(
+                "FAILED missingness check for %s: %.2f%% missing",
+                field,
+                100 * missingness,
+            )
+        else:
+            logger.info(
+                "Passed missingness check for %s: %.2f%% missing",
+                field,
+                100 * missingness,
+            )
+
+    logger.info("Checking for missingness of info and non-info fields...")
+    # Gather info and non-info metrics (or if doesn't exists, set to an empty list).
+    # substract missingness dict
+    # info_metrics = list(ht.row.info) - missingness_dict.keys() or []
+    # non_info_metrics = list(ht.row).remove("info") - missingness_dict.keys() or []
+    info_metrics = set(ht.row.info) - missingness_dict.keys()
```

Comment:
this won't work if `info` isn't in the ht, right? it would need to be something like `info_metrics = set(ht.row.info) - missingness_dict.keys() if 'info' in ht.row else []`